src/components/Card.jsx

```diff
@@ -34,28 +42,80 @@ const useStyles = makeStyles((theme) => {
     }
   };
 });
+
 function CardView({ data }) {
   const classes = useStyles();
   const navigate = useNavigate();
   const [modalData, setModalData] = useState({ open: false });
+  const [liked, setLiked] = useState(false);
+
   const handleModalClose = () => {
     setModalData({});
   };
   function handleCardAction() {
-    if (data.content?.length > 0) setModalData({ open: true, data });
+    if (data.mediaLink?.length > 0 || data.content_html) {
+      setModalData({ open: true, data });
+    }
   }
-  return (
-    <>
-      <Card className={classes.card}>
-        <CardActionArea onClick={handleCardAction}>
+
+  const handleLike = () => {
+    setLiked(!liked);
+  };
+
+  // Conditional rendering of CardMedia based on the data type
+  const renderCardMedia = () => {
+    switch (data.type) {
+      case "image":
+        return (
           <CardMedia
             height={200}
             component="img"
-            alt="Contemplative Reptile"
-            image={data?.backgroundImg}
-            title={data?.title}
+            alt={data.title}
+            image={data.backgroundImg || DefaultBgImage}
+            title={data.title}
           />
-        </CardActionArea>
+        );
+      case "video":
+      case "youtube":
+      case "audio":
+        // For video, youtube, and audio, use a placeholder image or a specific thumbnail
+        // and overlay a play icon to indicate action.
+        return (
+          <Box position="relative" display="flex" alignItems="center" justifyContent="center" height={200}>
+            <CardMedia
+              component="img"
+              alt={data.title}
+              image={data.backgroundImg || DefaultBgImage}
+              title={data.title}
+              style={{ height: "100%", width: "100%", position: "absolute" }}
+            />
+            <IconButton onClick={handleCardAction} sx={{ color: "white", zIndex: 1 }}>
+              <PlayCircleOutlineIcon fontSize="large" />
+            </IconButton>
+          </Box>
+        );
+      case "pdf":
+        // For PDFs, might directly use an image if provided or a default one
+        return (
+          <CardMedia
+            height={200}
+            component="img"
+            alt={data.title}
+            image={data.backgroundImg || DefaultBgImage}
+            title={data.title}
+          />
+        );
+      default:
+        // Fallback for other types or undefined type
+        return <CardMedia height={200} component="img" alt={data.title} image={DefaultBgImage} title={data.title} />;
+    }
+  };
```

Comment:
in renderCardMedia method ,  try to refactor to  duplicated code , for example CardMedia is repeated in many places.